bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/IndexPersistenceMgr.java

```diff
@@ -177,7 +177,35 @@ private FileInfo putFileInfo(Long ledger, byte masterKey[], File lf, boolean cre
      */
     private File getNewLedgerIndexFile(Long ledger, File excludedDir)
                     throws NoWritableLedgerDirException {
-        File dir = ledgerDirsManager.pickRandomWritableDir(excludedDir);
+        return getNewLedgerIndexFile(ledger, excludedDir, false);
```

Comment:
Can you explain situations where fallback is not allowed and where it is OK.
Index files are small, but still it can push dirs to 100% and eventually write can fail.
Looks like this change has *NO* Checks to see the fullness of the randomly picked dir. Any consequences ?